packages/next/src/cli/next-dev.ts

```diff
@@ -209,6 +209,13 @@ const nextDev: CliCommand = async (args) => {
     traceUploadUrl = args['--experimental-upload-trace']
   }
 
+  // TODO: remove in the next major version
+  if (config.analyticsId) {
+    Log.warnOnce(
```

Comment:
`warnOnce` in development is correct when it reruns, in this case I don't think `next-dev.ts` reruns so it can be `Log.warn` too:

```suggestion
    Log.warn(
```